feat: Add performance docs for react (#1792)
getsentry/sentry-docs: src/collections/_documentation/performance-monitoring/configuration/react.md

> To get started with performance monitoring using Sentry's React SDK, first install the `@sentry/react` and `@sentry/apm` packages:

Reviewer: Add header here:

```suggestion

**React**

To get started with performance monitoring using Sentry's React SDK, first install the `@sentry/react` and `@sentry/apm` packages:
```

```bash
# Using yarn
$ yarn add @sentry/react @sentry/apm

# Using npm
$ npm install @sentry/react @sentry/apm
```

Next, initialize the integration in your call to `Sentry.init`. Make sure this happens before you mount your React component on the page.

```jsx
import * as Sentry from '@sentry/react';
import { Integrations } from "@sentry/apm";
Sentry.init({
  dsn: '"___PUBLIC_DSN___"',
  release: 'my-project-name@' + process.env.npm_package_version,
  integrations: [
    new Integrations.Tracing(),
  ],
  tracesSampleRate: 0.25, // must be present and non-zero
});

// ...

ReactDOM.render(<App />, rootNode);

// Can also use with React Concurrent Mode
// ReactDOM.createRoot(rootNode).render(<App />);
```

> ### `withProfiler` Higher-Order Component

Reviewer: ```suggestion
**`withProfiler` Higher-Order Component**
```

`@sentry/react` exports a `withProfiler` higher-order component that leverages the `@sentry/apm` Tracing integration to add React related spans to transactions. If the Tracing integration is not enabled, the Profiler will not work.

In the example below, the `withProfiler` higher-order component is used to instrument an App component.

```jsx
import React from 'react';
import * as Sentry from '@sentry/react';

class App extends React.Component {
  render() {
    return (
      <FancyComponent>
        <NestedComponent someProp={2} />
        <AnotherComponent />
      </FancyComponent>
    )
  }
}

export default Sentry.withProfiler(App);
```

> The React Profiler current generates spans with three different kinds of op codes: `react.mount`, `react.render`, and `react.update`.

Reviewer 2: ```suggestion
The React Profiler currently generates spans with three different kinds of op-codes: `react.mount`, `react.render`, and `react.update`.
```

> 1. `react.mount` is the span that represents how long it took for the profiled component to mount.

Reviewer 2: ```suggestion
`react.mount`

: The span that represents how long it took for the profiled component to mount.
```

Reviewer 2: Looks like you're defining these op-codes. There's a stylistic way Docs uses to define terminology, components, etc. Trying to be more consistent about it. 😄 Made the changes for ya.

Author: Ok great, thanks for pointing that out, I'll start following that convention.

> 2. `react.render` is the span that represents how long the profiled component was on a page. This span is only generated if the profiled component mounts and unmounts while a transaction is occuring.

Reviewer 2: ```suggestion
`react.render`

: The span that represents how long the profiled component was on a page. This span is only generated if the profiled component mounts and unmounts while a transaction is occurring.
```

> 2. `react.update` is the span that represents when the profiled component updated. This span is only generated if the profiled component has mounted.

Reviewer 2: ```suggestion
`react.update`

: The span that represents when the profiled component updated. This span is only generated if the profiled component has mounted.
```

{% capture __alert_content -%}
In [React Strict Mode](https://reactjs.org/docs/strict-mode.html), certain component methods will be [invoked twice](https://reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects). This may lead to duplicate `react.mount` spans appearing in a transaction. React Strict Mode only runs in development mode, so this will have no impact on your production traces.
{%- endcapture -%}
{%- include components/alert.html
  title="Note"
  content=__alert_content
  level="warning"
%}

The `withProfiler` higher-order component has a variety of options for further customization. They can be passed in as the second argument to the `withProfiler` function.

```jsx
export default Sentry.withProfiler(App, { name: "CustomAppName" })
```

> #### `name` (string)

Reviewer 2: ```suggestion
`name` (string)
```

> The name of the component being profiled. By default, the name is taken from the component `displayName` property or the component `name` property.

Reviewer 2: ```suggestion
: The name of the component being profiled. By default, the name is taken from the component `displayName` property or the component `name` property.
```

Reviewer 2: Looks like you're defining these components. There's a stylistic way Docs uses to define terminology, components, etc. Trying to be more consistent about it. 😄 Made the changes for ya.

> #### `includeRender` (boolean)

Reviewer 2: ```suggestion
`includeRender` (boolean)
```

> If a `react.render` span should be created by the Profiler. Set as true by default.

Reviewer 2: ```suggestion
: If a `react.render` span should be created by the Profiler. Set as true by default.
```

> #### `includeUpdates` (boolean)

Reviewer 2: ```suggestion
`includeUpdates` (boolean)
```

> If `react.update` spans should be created by the Profiler. Set as true by default. We recommend setting this prop as false for components that will experience many rerenders, such as text input components, as the resulting spans can be very noisy.

Reviewer 2: ```suggestion
: If `react.update` spans should be created by the Profiler. Set as true by default. We recommend setting this prop as false for components that will experience many rerenders, such as text input components, as the resulting spans can be very noisy.
```
